### PR TITLE
Add unlearner to inverted_index_euclid

### PR DESCRIPTION
**Pull request by rimms**
This patch is a part of #189

### CODE REVIEW

**Comment by kmaehashi** [comment]
cpplint!

File: jubatus/core/recommender/inverted_index_euclid.cpp [hunk 1]
```diff
@@ -32,6 +32,11 @@ inverted_index_euclid::inverted_index_euclid() {
 inverted_index_euclid::~inverted_index_euclid() {
 }
 
+inverted_index_euclid::inverted_index_euclid(
+      jubatus::util::lang::shared_ptr<unlearner::unlearner_base> unlearner)
+  : inverted_index(unlearner){
```
Comment (kmaehashi) [inline jubatus/core/recommender/inverted_index_euclid.cpp]: cpplint violation.

File: jubatus/core/recommender/inverted_index_euclid.hpp [hunk 2]
```diff
@@ -30,6 +30,8 @@ class inverted_index_euclid : public inverted_index {
  public:
   inverted_index_euclid();
   ~inverted_index_euclid();
+  inverted_index_euclid(
+      jubatus::util::lang::shared_ptr<unlearner::unlearner_base> unlearner);
```
Comment (kmaehashi) [inline jubatus/core/recommender/inverted_index_euclid.hpp]: Single argument constructors must declare `explicit`.

**Comment by kmaehashi** [comment]
@rimms Reviewed.

**Comment by rimms** [comment]
@kmaehashi fixed.

**Comment by kmaehashi** [comment]
:+1:

**Pull request merged**